master/internal/resourcemanagers/job.go

```diff
@@ -0,0 +1,92 @@
+package resourcemanagers
+
+import (
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/jobv1"
+)
+
+// FIXME haven't decided if resource manager actor should be responsible for this or not
+// we don't want a separate actor do we? could be useful for streaming job endpoints
+type GetJobOrder struct{}
+
+func filterAllocateRequests(reqs AllocReqs) AllocReqs {
```

Comment:
clarification: does this just filter allocation requests for if their jobs exist?

Reply:
i should add a comment to the code
1. filters allocations that are not associated with a job
2. if a job has many allocReqs it would only keep the one that's most representative of the final job state. [more reading] with the current state of the schedulers we can achieve this without having to introduce a new struct eg doing additional computation for coming up with the desired `schedulingstate` w/o sacrificing anything.